streaming/src/queue/queue.cc

```diff
@@ -169,6 +168,115 @@ void WriterQueue::OnNotify(std::shared_ptr<NotificationMessage> notify_msg) {
   min_consumed_id_ = notify_msg->SeqId();
 }
 
+void WriterQueue::ResendItem(QueueItem &item, uint64_t first_seq_id,
+                             uint64_t last_seq_id) {
+  ResendDataMessage msg(actor_id_, peer_actor_id_, queue_id_, first_seq_id, item.SeqId(),
+                        item.MsgIdStart(), item.MsgIdEnd(), last_seq_id, item.Buffer(),
+                        item.IsRaw());
+  STREAMING_CHECK(item.Buffer()->Data() != nullptr);
+  std::unique_ptr<LocalMemoryBuffer> buffer = msg.ToBytes();
+
+  transport_->Send(std::move(buffer));
+}
+
+int WriterQueue::ResendItems(std::list<QueueItem>::iterator start_iter,
+                             uint64_t first_seq_id, uint64_t last_seq_id) {
+  std::unique_lock<std::mutex> lock(mutex_);
+  int count = 0;
+  auto it = start_iter;
+  for (; it != watershed_iter_; it++) {
+    if (it->SeqId() > last_seq_id) {
+      break;
+    }
+    STREAMING_LOG(INFO) << "ResendItems send seq_id " << (*it).SeqId() << " to peer.";
+    ResendItem(*it, first_seq_id, last_seq_id);
+    count++;
+  }
+
+  STREAMING_LOG(INFO) << "ResendItems total count: " << count;
+  is_resending_ = false;
+  return count;
+}
+
+void WriterQueue::FindItem(
+    uint64_t target_msg_id, std::function<void()> large_callback, std::function<void()> small_callback,
```

Comment:
It's confused to use callback like `large_callback` and `small_callback`.

Reply:
Better to use `greater` and `less`.